aiokafka/producer/producer.py

```diff
@@ -350,25 +346,7 @@ async def start(self):
         ), "Please create objects with the same loop as running with"
         log.debug("Starting the Kafka producer")  # trace
         await self.client.bootstrap()
-
-        if self._compression_type == "lz4":
-            assert self.client.api_version >= (0, 8, 2), (
```

Comment:
What behaviour do we have in that case now?

Reply:
Now, it will fails later in the process, when trying to actually send a query to the producer: 
* either the request class is raising a IncompatibleBroker exception, because it knows already that the provided parameters are not compatible with the negotiated version
* or the request is actually sent to the broker, that will return an error